dotnet/src/Connectors/Connectors.Memory.Postgres/PostgresVectorStoreCollectionSqlBuilder.cs

```diff
@@ -367,14 +376,24 @@ public PostgresSqlCommandInfo BuildGetNearestMatchCommand(
         };
 
         var vectorColumn = vectorProperty.StoragePropertyName ?? vectorProperty.DataModelPropertyName;
+
         // Start where clause params at 2, vector takes param 1.
-        var where = GenerateWhereClause(schema, tableName, properties, filter, startParamIndex: 2);
+#pragma warning disable CS0618 // VectorSearchFilter is obsolete
+        var (where, parameters) = (oldFilter: legacyFilter, newFilter) switch
+        {
+            (not null, not null) => throw new ArgumentException("Either Filter or NewFilter can be specified, but not both"),
```

Comment:
nit: use nameof(Filter) and nameof(NewFilter) in case params are renamed

Reply:
Good suggestion - I'll pass in this specific case only because Filter will be removed soon (before release) anyway and this whole piece of code will go away.